tests/core/polynomials_test.py

```diff
@@ -0,0 +1,329 @@
+                evaluate_polynomial_at_measuring_distances,
+            )
+
+            # Test for a simple quadratic polynomial: 2 + 3x + 4x^2
+            distances: ndarray = array([1, 2, 3])
+            coefficients: ndarray = array([2, 3, 4])
+            # At x = 1: 2 + 3(1) + 4(1^2) = 9
+            # At x = 2: 2 + 3(2) + 4(2^2) = 2 + 6 + 16 = 24
+            # At x = 3: 2 + 3(3) + 4(3^2) = 2 + 9 + 36 = 47
+            expected_result: ndarray = array([9, 24, 47])
+            testing.assert_array_equal(
+                evaluate_polynomial_at_measuring_distances(distances, coefficients),
+                expected_result,
+            )
+
+    @staticmethod
+    def test_CanEvaluateAPolynomialAtMeasuringDistancesForEmptyDistanceInput():
+        """Can evaluate a polynomial at measuring distances for empty distances."""
+        numpy_module_mock = set_up_mocks()
+
+        # Mocked return values of called functions
+        numpy_module_mock.power.side_effect = [array([]), array([]), array([])]
+        numpy_module_mock.zeros_like.return_value = array([])
+
+        with patch.dict(
+            "sys.modules",
+            {
+                "numpy": numpy_module_mock,
+            },
+        ):
+            from napytau.core.polynomials import (
+                evaluate_polynomial_at_measuring_distances,
+            )
+
+            distances: ndarray = array([])
+            coefficients: ndarray = array([2, 3, 4])
+            # With an empty input array, the result should also be an empty array
+            expected_result: ndarray = array([])
+            testing.assert_array_equal(
+                evaluate_polynomial_at_measuring_distances(distances, coefficients),
+                expected_result,
+            )
+
+    @staticmethod
+    def test_CanEvaluateAPolynomialAtMeasuringDistancesForSingleDistanceMeasurement():
+        """Can evaluate a polynomial at measuring distances for single distance."""
+        numpy_module_mock = set_up_mocks()
+
+        # Mocked return values of called functions
+        numpy_module_mock.power.side_effect = [array([1]), array([2]), array([4])]
+        numpy_module_mock.zeros_like.return_value = array([0])
+
+        with patch.dict(
+            "sys.modules",
+            {
+                "numpy": numpy_module_mock,
+            },
+        ):
+            from napytau.core.polynomials import (
+                evaluate_polynomial_at_measuring_distances,
+            )
+
+            distances: ndarray = array([2])
+            coefficients: ndarray = array([1, 2])
+            # Polynomial: f(x) = 1 + 2x
+            # At x = 2: 1 + 2(2) = 5
+            expected_result: ndarray = array([5])
+            testing.assert_array_equal(
+                evaluate_polynomial_at_measuring_distances(distances, coefficients),
+                expected_result,
+            )
+
+    @staticmethod
+    def test_CanEvaluateASingleValuePolynomialAtMeasuringDistances():
```

Comment:
This is probably trying to say, a polynomial of degree one. The function always runs on a single polynomial